packages/itwin/grouping-mapping-widget/src/widget/components/customUI/GroupingMappingCustomUI.ts

```diff
@@ -16,20 +26,68 @@ export interface GroupingMappingCustomUI {
    * Display label in the widget.
    */
   displayLabel: string;
-  /**
-   * Custom UI Component to build query interactively. Refer to searchGroupingUI/manualGroupingUI.
-   */
-  uiComponent: (props: GroupingMappingCustomUIProps) => JSX.Element;
   /**
    * Optional icon, will be shown before display label in widget.
    */
   icon?: ReactElement;
 }
 
 /**
- * Customized UI component arguments definition.
+ * Context custom UI type definition.
+ */
+export interface ContextCustomUI extends IGroupingMappingCustomUI {
+  /**
+   * UI type.
+   */
+  type: GroupingMappingCustomUIType.CONTEXT;
+  /**
+   * User defined component for UI interaction with grouping mapping widget.
+   */
+  uiComponent?: (props: ContextCustomUIProps) => JSX.Element;
```

Comment:
It would be great to have an example of such `ContextCustomUI`. Maybe you could try and think of something?